docs/manual/resource-leak-checker.tex

```diff
@@ -0,0 +1,343 @@
+\item \<p> is stored in an \<@Owning> field of the enclosing class.
+\end{enumerate}
+
+Support for \<@MustCallAlias> can be disabled by passing the \<-AnoResourceAliases> command-line
+argument to the checker. In this mode, resource-alias sets are not tracked by the checker,
+and \<@MustCallAlias> is treated identically to \<@PolyMustCall>.
+
+
+\sectionAndLabel{Creating obligations (how to re-assign a non-final owning field)}{resource-leak-createsmustcallfor}
+
+Consider a class that has must-call obligations; that is, the class
+declaration is annotated with \<@MustCall(...)>.
+Every constructor implicitly creates obligations for the newly-created object.
+Non-constructor methods may also create obligations
+when re-assigning non-final owning fields or allocating
+new system-level resources.
+
+A post-condition annotation,
+\<@CreatesMustCallFor>,
+indicates for which expression an obligation is created.
+If you write \<@CreatesMustCallFor(>\emph{T}\<)> on a method \emph{M} that
+overrides a method \emph{N}, then \emph{N} must also be annotated as
+\<@CreatesMustCallFor(>\emph{T}\<)>.  (\emph{N} may also have other
+\<@CreatesMustCallFor> annotations that \emph{M} does not.)
+If you write \<@CreatesMustCallFor> on a method that does not actually
+create an obligation, the checker may issue false positive warnings at calls to
+that method, but it never causes
+missed warnings (i.e., the checker is sound).
+
+\<@CreatesMustCallFor> allows the Resource Leak Checker to verify uses of non-final fields
+that contain a resource, even if they are re-assigned. Consider
+the following example:
+
+\begin{verbatim}
+  @MustCall("close") // default qualifier for uses of SocketContainer
+  class SocketContainer {
+    private @Owning Socket sock;
+
+    public SocketContainer() { sock = ...; }
+
+    void close() { sock.close() };
+
+    @CreatesMustCallFor("this")
+    void reconnect() {
+      if (!sock.isClosed()) {
+        sock.close();
+      }
+      sock = ...;
+    }
+  }
+\end{verbatim}
+
+In the lifetime of a \<SocketContainer> object, \<sock>
+might be re-assigned arbitrarily many times: once at each
+call to \<reconnect()>. This code is safe, however: \<reconnect()>
+ensures that \<sock> is closed before re-assigning it.
+
+Support for \<@CreatesMustCallFor> can be disabled by passing the \<-AnoCreatesMustCallFor> command-line
+argument to the checker. In this mode, \<@CreatesMustCallFor> annotations on methods are ignored at call-sites,
+and re-assigning a non-final, owning field is always an error.
+
+Sections~\ref{resource-leak-createsmustcallfor-callsite}
+and~\ref{resource-leak-createsmustcallfor-declaration}
+explain how the Resource Leak Checker verifies uses and declarations of
+methods annotated with \<@CreatesMustCallFor>.
+
+
+\subsectionAndLabel{Requirements at a call site of a \<@CreatesMustCallFor> method}{resource-leak-createsmustcallfor-callsite}
+
+At a call site to a method annotated as
```

Comment:
The structure of section 10.5 was not clear.
"At a call site": this talks about half of rely-guarantee reasoning.  Where is the other part?  It's in section 10.5.1, but that isn't clear, and also not from the title of section 10.5.1.
Section 10.5.1 has two unrelated parts: an example, and the requirements.

I suggest reorganizing:
 * Put the example early, before the discussion of requirements.  It is generally useful to readers.
 * State that there are requirements at the call site and at the declaration.
 * Have a subsection for requirements at the call site.
 * Have a subsection (this one already exists) for requirements at the declaration.

I took a first pass at this, but it needs a careful re-reading and editing.